### DISCUSSION TITLE
Static Html Export: How fix asset paths? (CSS, custom fonts, images)

### DISCUSSION THREAD

**Discussion by aaRitsch** [vercel/next.js#43867] [Help]
I created a next project using `next build && next export`. When i open the index.html file from the 'out' folder in a browser assets were not loaded. Not css, images and custom fonts. Because of wrong asset paths.

The exported paths are for example: `/vercel.svg` but should be `./vercel.svg` to load properly.

The problem is that the latter path gives an error on `npm run dev`, so can't be changed in the repo.

What is the recommended way to use `next export` which leads to correct asset paths?

The [Static HTML Export](https://nextjs.org/docs/advanced-features/static-html-export) documentation does not mention any requirements for configuration.

I already found out that adding `assetPrefix: './'` or `assetPrefix: '.'` to next.config.js makes CSS load correctly. But not custom fonts and images.

But is that the recommended way to solve the problem partly? 

What approach should be used to solve the custom font and image paths?

I am aware of another workaround which works good for images: Changing paths by environment variable: `process.env.NODE_ENV === 'development' ? '/vercel.svg' : './vercel.svg'`

There are some discussions about this topic out there, but I couldn't find a satisfying answer.

Any recommendations?

Here is an example to work with:
https://github.com/aaRitsch/next-build-static-asset-path

**Comment by GabenGar** [comment]
The paths are not "wrong", they are assumed to be ran on a server instead of a file.

**Reply by aaRitsch** [reply]
What exactly do you mean by "ran on a server"? Do you mean self-hosted servers with Node.js or Docker Image as described here? https://nextjs.org/docs/deployment

I thought "static html export" is an alternative to this approach.

**Reply by GabenGar** [reply]
Static html export only creates static files so they could be used on servers other than nextjs/vercel ones. It's not even possible to make it work in `file://` links, since the browser won't let you make fetch requests from them.

**Comment by Yarles404** [comment]
I had this same issue, and the solutions feel like janky workarounds. Asset paths have no business being absolute paths.

**Reply by aaRitsch** [reply]
Do you have recommendations for better workarounds? Did you find a workaround which solves the custom font paths?
